### ISSUE TITLE
Improved macro support for HTML frame/dialog/overlay creation

### ISSUE THREAD

**Issue by cwisniew** [RPTools/maptool#2964]
**Is your feature request related to a problem? Please describe.**
Add macros to open frame/frame5/dialog/dialog5/overlays that don't require the HTML page to be encoded in the macro or pulled in using some other trick like
```
[frame5("test"): {
	[r: getMacroCommand(number(getMacroIndexes("test/test.html"))) ]
}]
```
This should also be compatible/consistent with #2963

**Comment by cwisniew** [comment]
Adds url scheme to fetch from lib:tokens which can be used in the function for creating HTML frame/frame5/dialog/dialog5/overlay

```
html.dialog(name, liburl [, options ]) 
html.dialog5(name, liburl [, options ]) 
html.frame(name, liburl [, options ]) 
html.frame5(name, liburl [, options ]) 
html.overlay(name, liburl [, options ]) 
```
The optional "options" argument contains the same arguments and format as the frame/dialog/overlay roll options. 

Note: The roll options do not support the new URL scheme so there is no risk of breaking those. 

Scheme is
- lib://*&lt;tokename&gt;*/macro/&lt;macro name&gt;
- lib://*&lt;tokenname&gt;*/property/*&lt;property name&gt;*

**Things to note:**
- *&lt;tokenname&gt;* is the name of the lib:token without the "lib:" part.
- The macro name can contain / characters making it appear to be a directory path + file


See attached campaign file for examples
[test-fw-ext-cmpgn.zip](https://github.com/RPTools/maptool/files/7147873/test-fw-ext-cmpgn.zip)

**Comment by Phergus** [comment]
Leaving open for now.

**Comment by Phergus** [comment]
Error message from bad URL (or checkbox not checked) doesn't include the URL info:
```
Invalid URL {0}.
Error trace : test - frame5@Lib:Test
```

**Comment by Phergus** [comment]
Tested fix from#2968.  Error message now returns the offending URI.

**Comment by Phergus** [comment]
Tested new macro functions.  Working as expected.
